microsoft/devhome
# Made the Insights button more standard, more predictable, more useful

## Summary of the pull request
Up to now, when we get a report in the Windows Logs page that we can analyze and suggest mitigations for, we add a new Insight to the Insights page, and draw attention to this by animating the Insights button on the Windows Logs page. I've updated this to use standard button with an InfoBadge instead, and removed the custom GlowButton control altogether.


## Detailed description
Per our guiding principles, I've made it so this button is always available (doesn't appear/disappear as it did before), and moved it to a standard location in the main bar. As part of this, added the BarWindowViewModel as a singleton to the app's services collection.

The InfoBadge numeric value is bound to the count of unread Insights items, and I updated the Insights class to expose a HasBeenRead property for this. I'm databinding the count of unread insights to the InfoBadge numeric value. I'd like to use this same property to bind to the Opacity, via a converter. However, while WinUI has the problem that we can't use a binding converter for a WinUI Window type https://github.com/microsoft/microsoft-ui-xaml/issues/4966, I'll stick with the approach of declaring a second dependent property calculated in the viewmodel instead. I'm using Opacity rather than Visibility because changing Opacity doesn't change the layout.

In the InsightsPage, each Insight is bound to an Expander in the UI, which defaults to Collapsed. When the user expands an item, this sets the Insights.HasBeenRead property to true. Subsequent collapsing/expanding doesn't change the value - this is ensured via a custom Expander behavior. I've also added an InfoBadge to each Expander - this is a simple non-numeric badge to tell the user this item has not yet been read. When she expands the item, the InfoBadge is hidden, and again any subsequent collapse/expand doesn't change this.

Note: when we complete https://github.com/microsoft/devhome/issues/3063 (Reimplement PI's main nav list to use NavigationView), we should also move the InfoBadge to the Insights nav item (added a note in that issue).

## Validation steps performed
1 Run PI and associate with ProblemApp2.
2 Run AnotherApp, and lock the shared file.
3 Expand PI and go to the Windows Logs page.
4 In ProblemApp2, tap "Locked file access".
5 See a new report shows up in the Windows Logs page, and a new Insight in the Insights page.
6 Also see the numeric info badge appears on the Insights button in the bar.
7 Repeat #4 several times, see the numeric value increases.
8 Click the Insights button (or the Insights nav item) to go to the Insights page.
9 See each unread/unexpanded item has a badge.
10 Expand any item(s) to read the content.
11 See the expander badge goes away; also see value in the numeric badge in the bar is decremented.
12 Collapse/re-expand any item(s), and see that neither badge changes any more.
13 Detach from the target app.
14 See the Insights and badges remain (by design), and expanding unread items behaves as before.
15 Associate to a different app.
16 See the Insights are cleared, and the numeric badge in the bar is removed.
17 Associate to ProblemApp2 again: see that this starts fresh, nothing left over from last time.

## PR checklist
Closes #3218
Closes #3469

https://github.com/user-attachments/assets/b2ca36f1-1124-46d1-bf29-ebd8700d727d
